tests/test_tasks.py

```diff
@@ -995,6 +996,64 @@ def bar():
         assert second_state.name == "Completed"
         assert second_state.result() != first_state.result()
 
+    def test_cache_misses_w_refresh_cache(self):
+        @task(cache_key_fn=lambda *_: "cache hit", refresh_cache=True)
+        def foo(x):
+            return x
+
+        @flow
+        def bar():
+            return foo._run(1), foo._run(2)
+
+        first_state, second_state = bar()
+        assert first_state.name == "Completed"
+        assert second_state.name == "Completed"
+        assert second_state.result() != first_state.result()
+
+    def test_cache_hits_wo_refresh_cache(self):
+        @task(cache_key_fn=lambda *_: "cache hit", refresh_cache=False)
+        def foo(x):
+            return x
+
+        @flow
+        def bar():
+            return foo._run(1), foo._run(2)
+
+        first_state, second_state = bar()
+        assert first_state.name == "Completed"
+        assert second_state.name == "Cached"
+        assert second_state.result() == first_state.result()
+
+    def test_tasks_refresh_cache_setting(self):
+        @task(cache_key_fn=lambda *_: "cache hit")
+        def foo(x):
+            return x
+
+        @task(cache_key_fn=lambda *_: "cache hit", refresh_cache=True)
+        def refresh_task(x):
+            return x
+
+        @task(cache_key_fn=lambda *_: "cache hit", refresh_cache=False)
+        def not_refresh_task(x):
+            return x
+
+        @flow
+        def bar():
+            foo.submit(0).wait
+            return (
+                foo.submit(1).wait(),
+                refresh_task.submit(2).wait(),
+                not_refresh_task.submit(3).wait(),
+            )
+
+        with temporary_settings({PREFECT_TASKS_REFRESH_CACHE: True}):
```

Comment:
This test should fail per https://github.com/PrefectHQ/prefect/pull/7856/files#r1062934837 — does it?

Reply:
You are right; it failed. As I moved the decision task argument vs. Prefect setting to runtime, it is now working as expected and not failing anymore.